tests/models/timesformer/test_modeling_timesformer.py

```diff
@@ -0,0 +1,349 @@
+
+
+class TimeSformerModelTester:
+    def __init__(
+        self,
+        parent,
+        batch_size=13,
+        image_size=10,
+        num_channels=3,
+        patch_size=2,
+        num_frames=2,
+        is_training=True,
+        use_labels=True,
+        hidden_size=32,
+        num_hidden_layers=5,
+        num_attention_heads=4,
+        intermediate_size=37,
+        hidden_act="gelu",
+        hidden_dropout_prob=0.1,
+        attention_probs_dropout_prob=0.1,
+        type_sequence_label_size=10,
+        initializer_range=0.02,
+        attention_type="divided_space_time",
+        scope=None,
+    ):
+        self.parent = parent
+        self.batch_size = batch_size
+        self.image_size = image_size
+        self.num_channels = num_channels
+        self.patch_size = patch_size
+        self.num_frames = num_frames
+        self.is_training = is_training
+        self.use_labels = use_labels
+        self.hidden_size = hidden_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.intermediate_size = intermediate_size
+        self.hidden_act = hidden_act
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.attention_probs_dropout_prob = attention_probs_dropout_prob
+        self.attention_type = attention_type
+        self.initializer_range = initializer_range
+        self.scope = scope
+        self.type_sequence_label_size = type_sequence_label_size
+
+        # in TimeSformer, the number of spatial tokens equals num_frames * num_patches per frame + 1 CLS token
+        self.num_patches_per_frame = (image_size // patch_size) ** 2
+        self.seq_length = (num_frames) * self.num_patches_per_frame + 1
+
+    def prepare_config_and_inputs(self):
+        pixel_values = floats_tensor(
+            [self.batch_size, self.num_frames, self.num_channels, self.image_size, self.image_size]
+        )
+
+        labels = None
+        if self.use_labels:
+            labels = ids_tensor([self.batch_size], self.type_sequence_label_size)
+
+        config = self.get_config()
+
+        return config, pixel_values, labels
+
+    def get_config(self):
+        return TimeSformerConfig(
+            image_size=self.image_size,
+            patch_size=self.patch_size,
+            num_channels=self.num_channels,
+            num_frames=self.num_frames,
+            hidden_size=self.hidden_size,
+            num_hidden_layers=self.num_hidden_layers,
+            num_attention_heads=self.num_attention_heads,
+            intermediate_size=self.intermediate_size,
+            hidden_act=self.hidden_act,
+            hidden_dropout_prob=self.hidden_dropout_prob,
+            attention_probs_dropout_prob=self.attention_probs_dropout_prob,
+            initializer_range=self.initializer_range,
+            attention_type=self.attention_type,
+        )
+
+    def create_and_check_model(self, config, pixel_values, labels):
```

Comment:
Please also add a check for the video classifier head model

Reply:
Sorry but may I ask what would be different compared to the checks that are implemented here: https://github.com/huggingface/transformers/blob/2fb06b332142009f140c6bfb3d040a397ea6312e/tests/models/timesformer/test_modeling_timesformer.py#L330-L349

Reply:
The test you're referring to is the integration test, but we also have tests that just verify the shapes of the last hidden states/logits, so it'd be great to add a `create_and_check_for_video_classification` method and a corresponding `test_for_video_classification` test, see VideoMAE as an example

Reply:
I understood 👍 Added `test_for_video_classification` and `create_and_check_for_video_classification`.